### DISCUSSION TITLE
Reference to SW for HW ResTouch-LCD-2.8

### DISCUSSION THREAD

**Discussion by JoGeDuBo** [micropython/micropython#16194] [RP2040 / Pico]
With a little adaptation, this excellent SW https://github.com/unital/tempe runs on this HW https://www.waveshare.com/wiki/Pico-ResTouch-LCD-2.8 too. 
(I myself, as a hobby, had already started to write a similar, but a few numbers smaller, SW for this HW with some success. I probably don't need to pursue this any further now. ) 
Here https://github.com/unital/tempe/blob/main/src/tempe_displays/st7789/pimoroni.py only this
```
class WaveshareDisplay(PimoroniDisplay):
    def __init__(
            self,
            spi=None,
            cs_pin=17,
            dc_pin=16,
            backlight_pin=20,
            size=(320, 240),
            centered=False,
            reset_pin=None,
        ): # = c&p from PimoroniDisplay
        super().__init__(
            spi=SPI(1, baudrate=62_500_000, phase=1, polarity=1, sck=Pin(10, Pin.OUT), mosi=Pin(11, Pin.OUT)),
            cs_pin=9,
            dc_pin=8,
            backlight_pin=13,
            reset_pin=15,
        )

PimoroniDisplay = WaveshareDisplay # needed for the examples
```
has to be inserted at the end of the file, then all examples from https://github.com/unital/tempe/tree/main/example will run (except temperature_screen.py). 

In PimoroniDisplay.__init__, line 44, this code should be inserted for the sake of completeness, even if reset is not used in the examples:
```
if isinstance(reset_pin, int): reset_pin = Pin(reset_pin, Pin.OUT)
```

**Comment by corranwebster** [comment]
Excellent! I'm glad you found the library useful and could get it work on your hardware.  Any other feedback you might have would be very appreciated.

Thanks for this information - I've fixed that here: https://github.com/unital/tempe/pull/58

If it's OK with you I will make a waveshare class based on the settings you have here.

And yes, I need to update the temperature sensor example, but I didn't have an issue open for it.  I do now: https://github.com/unital/tempe/issues/59
